Source/ARTConnection.m

```diff
@@ -284,6 +262,32 @@ - (void)off:(ARTEventListener *)listener {
     [_eventEmitter off:listener];
 }
 
+-(NSString *)recoveryKey{
+    return [self createRecoveryKey];
+}
+
+- (NSString *)createRecoveryKey {
+    // RTN16h
+    if (_state == ARTRealtimeClosing
+        || _state == ARTRealtimeClosed
+        || _state == ARTRealtimeFailed
+        || _state == ARTRealtimeSuspended
+        || (!_key && !_id)) {
+        return nil;
+    }
+  
+    NSMutableDictionary<NSString *,NSString *> *serials = @{}.mutableCopy;
```

Comment:
```suggestion
    NSMutableDictionary<NSString *,NSString *> *serials = [NSMutableDictionary new];
```

Reply:
`@{}.mutableCopy` just looks ugly IMHO =)

Reply:
And, technically, creates two objects.